app/models/support_interface/vendor_api_monitor.rb

```diff
@@ -0,0 +1,66 @@
+module SupportInterface
+  class VendorAPIMonitor
+    def never_connected
+      @_never_connected = providers.left_outer_joins(:vendor_api_requests)
+        .includes(:vendor_api_tokens)
+        .distinct
+        .where(vendor_api_requests: { id: nil })
+    end
+
+    def no_sync_in_24h
+      @_no_sync_in_24h ||= connected_providers.where.not(
+        id: VendorAPIRequest.syncs.select(:provider_id).distinct.where('created_at > ?', 24.hours.ago),
+      ).map do |provider|
+        ProviderWithAPIUsageStats.new(provider)
```

Comment:
The use of `ProviderWithAPIUsageStats` this way is causing similar queries for each provider. Although locally its not really an issue I think on production with a large number of vendor API requests it will be. Is there a way we can avoid querying the `VendorAPIRequest` twice, once at the higher level to retrieve the provider, and then again through the provider. Maybe we can have one query for each type of monitor instead? Appreciate this might be a bit of restructuring. Let me know your thoughts

Reply:
as discussed, my SQL is a bit weak rn — we can iterate this!